fk/lisjp/src/main/java/se/inera/intyg/common/lisjp/v1/model/converter/InternalToCertificate.java

```diff
@@ -681,4 +772,866 @@ private static CertificateDataValueDiagnosis createDiagnosis(String id, Diagnos
+                                singleExpression(PrognosTyp.ATER_X_ANTAL_DGR.getId()),
+                                singleExpression(PrognosTyp.SANNOLIKT_EJ_ATERGA_TILL_SYSSELSATTNING.getId()),
+                                singleExpression(PrognosTyp.PROGNOS_OKLAR.getId())
+                            )
+                        )
+                        .build(),
+                    CertificateDataValidationHide.builder()
+                        .questionId(AVSTANGNING_SMITTSKYDD_SVAR_ID_27)
+                        .expression(singleExpression(AVSTANGNING_SMITTSKYDD_SVAR_JSON_ID_27))
+                        .build()
+                }
+            )
+            .build();
+    }
+
+    private static CertificateDataElement createPrognosTimeperiodQuestion(LisjpUtlatandeV1 internalCertificate, int index) {
+        return CertificateDataElement.builder()
+            .id(PROGNOS_BESKRIVNING_DELSVAR_ID_39)
+            .index(index)
+            .parent(BEDOMNING_CATEGORY_ID)
+            .config(
+                CertificateDataConfigDropdown.builder()
+                    .list(
+                        Arrays.asList(
+                            DropdownItem.builder()
+                                .id("VALJ_TIDSPERIOD")
+                                .label("Välj tidsperiod")
+                                .build(),
+                            DropdownItem.builder()
+                                .id(PrognosDagarTillArbeteTyp.DAGAR_30.getId())
+                                .label(PrognosDagarTillArbeteTyp.DAGAR_30.getLabel())
+                                .build(),
+                            DropdownItem.builder()
+                                .id(PrognosDagarTillArbeteTyp.DAGAR_60.getId())
+                                .label(PrognosDagarTillArbeteTyp.DAGAR_60.getLabel())
+                                .build(),
+                            DropdownItem.builder()
+                                .id(PrognosDagarTillArbeteTyp.DAGAR_90.getId())
+                                .label(PrognosDagarTillArbeteTyp.DAGAR_90.getLabel())
+                                .build(),
+                            DropdownItem.builder()
+                                .id(PrognosDagarTillArbeteTyp.DAGAR_180.getId())
+                                .label(PrognosDagarTillArbeteTyp.DAGAR_180.getLabel())
+                                .build(),
+                            DropdownItem.builder()
+                                .id(PrognosDagarTillArbeteTyp.DAGAR_365.getId())
+                                .label(PrognosDagarTillArbeteTyp.DAGAR_365.getLabel())
+                                .build()
+                        )
+                    )
+                    .build()
+            )
+            .value(
+                CertificateDataValueCode.builder()
+                    .id(internalCertificate.getPrognos() != null && internalCertificate.getPrognos().getDagarTillArbete() != null
```

Comment:
Consider refactoring for cleaner code. Extracting method that creates a value if not null.